cpp/src/text/minhash.cu

```diff
@@ -74,20 +74,20 @@ struct minhash_fn {
     }
     __syncwarp();
 
-    auto const begin = d_str.begin() + lane_idx;
-    auto const end   = [d_str, width = width] {
-      auto const length = d_str.length();
-      if (length > width) { return (d_str.end() - (width - 1)); }
-      return d_str.begin() + static_cast<cudf::size_type>(length > 0);
-    }();
-
-    // each lane hashes substrings of the given width
-    for (auto itr = begin; itr < end; itr += cudf::detail::warp_size) {
-      auto const offset = itr.byte_offset();
-      auto const hash_str =
-        cudf::string_view(d_str.data() + offset, (itr + width).byte_offset() - offset);
+    auto const begin = d_str.data() + lane_idx;
+    auto const end   = d_str.data() + d_str.size_bytes();
```

Comment:
Could we still reduce `end` by `width` (or so) ? In which case do we need to hash a substring that starts after `d_str.data() + d_str.size_bytes() - width`?
To me it looks like we would bail at line 87 in such case.

Reply:
Unfortunately, `width` is in character units and `size_bytes()` is in bytes so this subtraction would be incorrect if the string contained multi-byte UTF8 characters near the end of the string. Subtracting `width` characters from the end of the string requires some amount of code logic so the bail out approach here is a bit more efficient.

Reply:
Surprising that does would be slower. Thanks for the explanation.